packages/slice-machine/src/components/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -24,11 +36,68 @@ export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
     >
       <DialogHeader title="Generate with AI" />
       <DialogContent gap={0}>
-        <Box height="100%" />
+        <Box padding={16} height="100%">
+          <FileDropZone
+            onFilesSelected={onFilesSelected}
+            assetType="image"
+            overlay={
+              <UploadBlankSlate
+                onFilesSelected={onFilesSelected}
+                droppingFiles
```

Comment:
this feels a bit weird, having the same component rendered twice, one as a child and another as an "overlay", with different configurations, rather than having a render prop coming from the file drop zone, is this how they implemented it in the page builder?

Reply:
I agree that it feels weird. In the page builder, the overlay is different from the blank state, so they don't have this issue.

Reply:
i've taken a look at the implementation and yeah, it's super weird, also because the "overlay" is only rendered while dropping files, but at the same time, nothing in its name indicates this